### PR TITLE
[SPARKR-154] Phase 4: implement subtract() and subtractByKey().

### PR DESCRIPTION
**Pull request by sun-rui**

### CODE REVIEW

**Comment by shivaram** [comment]
@concretevitamin Could you take a look ?

File: pkg/R/generics.R [hunk 1]
```diff
@@ -329,12 +329,24 @@ setGeneric("cogroup",
...
+setGeneric("sortByKey",
+           function(x, ascending = TRUE, numPartitions = 1L) {
+             standardGeneric("sortByKey")
+           })
 
+#' @rdname subtractByKey
+#' @export
+setGeneric("subtractByKey", 
+           function(x, other, numPartitions = 1L) {
+            standardGeneric("subtractByKey")
```
Comment (concretevitamin) [inline pkg/R/generics.R]: indent is 1 space off

File: pkg/R/generics.R [hunk 2]
```diff
@@ -329,12 +329,24 @@ setGeneric("cogroup",
...
+           })
+
+#' @rdname subtract
+#' @export
+setGeneric("subtract", 
+           function(x, other, numPartitions = 1L) {
+             standardGeneric("subtract")
```
Comment (concretevitamin) [inline pkg/R/generics.R]: indent is 1 space off

**Comment by concretevitamin** [comment]
Thanks @sun-rui - will merge pending indent nits and can you rebase off/merge master?

File: pkg/R/generics.R (hunk 1 above)
Comment (sun-rui) [inline pkg/R/generics.R]: fixed.

File: pkg/R/generics.R (hunk 2 above)
Comment (sun-rui) [inline pkg/R/generics.R]: The indent of this line is correct.

**Comment by sun-rui** [comment]
@concretevitamin  merged with master.

File: pkg/R/generics.R (hunk 2 above)
Comment (concretevitamin) [inline pkg/R/generics.R]: Yep, missed that under a coffee shop's dim light :)

**Comment by concretevitamin** [comment]
Thanks!

Merging into master & cherry-picking into sparkr-sql.

**Pull request merged**